### PR TITLE
[Toolchain][Parser] Add support for `package` directive.

### PR DESCRIPTION
**Pull request by ergawy**
Summary:

Adds parsing support for the `package` directive as specified by the
`Code and name organization` design doc.

### CODE REVIEW

File: toolchain/parser/parse_node_kind.def [hunk 1]
```diff
@@ -17,6 +17,13 @@
 CARBON_PARSE_NODE_KIND(DeclarationEnd)
 CARBON_PARSE_NODE_KIND(EmptyDeclaration)
 CARBON_PARSE_NODE_KIND(DeclaredName)
+
+CARBON_PARSE_NODE_KIND(PackageDirective)
+CARBON_PARSE_NODE_KIND(PackageApi)
+CARBON_PARSE_NODE_KIND(PackageImpl)
+CARBON_PARSE_NODE_KIND(PackageLibrary)
+CARBON_PARSE_NODE_KIND(PackageSemi)
```
Comment (jonmeow) [inline toolchain/parser/parse_node_kind.def]: Generally nodes are just "End", like StatementEnd
```suggestion
CARBON_PARSE_NODE_KIND(PackageEnd)
```

File: toolchain/parser/parser_impl.cpp [hunk 2]
```diff
@@ -448,6 +448,81 @@ auto ParseTree::Parser::ParseCodeBlock() -> llvm::Optional<Node> {
...
+
+  auto api_or_impl_token = tokens_.GetKind(*(position_));
+
+  if (api_or_impl_token == TokenKind::Api()) {
+    AddLeafNode(ParseNodeKind::PackageApi(), Consume(TokenKind::Api()));
+  } else if (api_or_impl_token == TokenKind::Impl()) {
+    AddLeafNode(ParseNodeKind::PackageImpl(), Consume(TokenKind::Impl()));
+  } else if (api_or_impl_token == TokenKind::StringLiteral()) {
+    CARBON_DIAGNOSTIC(MissingLibraryKeyword, Error,
+                      "Missing `library` keyword.");
```
Comment (jonmeow) [inline toolchain/parser/parser_impl.cpp:510]: You might want to add a test like:

```
package Foo library "bar" "baz";
```

I understand the utility of this message -- maybe a bool to track whether a library was already found?

**Review by jonmeow — COMMENTED** [review]
Just darting in with a couple comments while I'm glancing around, others can feel free to finish review (I don't think I'm the most authoritative here, and I'm still somewhat unavailable this week)

**Comment by jonmeow** [comment]
Also -- this change is looking good, thank you. I think my comments are pretty minor. :)

File: toolchain/parser/parser_impl.cpp (hunk 2 above)
Comment (ergawy) [reply toolchain/parser/parser_impl.cpp:510]: Ah, good catch, thanks! Done.

**Review by jonmeow — APPROVED** [review]
Thanks for the edits! I think others have had time to comment if they want to, so I'm just going to merge. 👍

**Pull request merged**